prow/cmd/hmac/main.go

```diff
@@ -425,6 +425,7 @@ func (c *client) batchOnboardNewTokenForRepos() error {
 			} else {
 				delete(c.currentHMACMap, repo)
 			}
+			return err
```

Comment:
Erroring out early here rather than completing the onboarding for other tokens is problematic. If the onboarding isn't completed for some reason we need to revert the hmac changes.

Rather than erroring out early I'd expect the error to be recorded, but for execution to continue. We should only report this error after cleanup has completed. The same applies to any possible validation errors from line 416.

Reply:
No worries.  I'll make a fix PR in a bit.

Reply:
https://github.com/kubernetes/test-infra/pull/21599